tests/test_sequence.py

```diff
@@ -996,13 +955,13 @@ def test_hardware_constraints():
     mid_delay = 40
     seq.delay(mid_delay, "ch0")
     seq.add(const_pls, "ch0")  # Phase = π
-    assert seq._last("ch0").ti - tf_ == rydberg_global.phase_jump_time
+    interval = seq._schedule["ch0"].adjust_duration(
+        rydberg_global.phase_jump_time + black_pls.fall_time(rydberg_global)
+    )
+    assert seq._last("ch0").ti - tf_ == interval
```

Comment:
This might not be ideal, since it is testing the behavior of `seq` against itself (if `adjust_duration` and `seq._last("ch0").ti - tf_` are both wrong, it may still pass).

Reply:
Changed to this:
```suggestion
    interval = seq._schedule["ch0"].adjust_duration(
        rydberg_global.phase_jump_time + black_pls.fall_time(rydberg_global)
    )
    assert seq._schedule["ch0"][-1].ti - tf_ == interval
```

Reply:
I would say that `seq` is still tested against itself here (between `seq._schedule["ch0"].adjust_duration` and `seq._schedule["ch0"][-1].ti`).

Reply:
Ok, how about this then? 
```python
    raw_interval = rydberg_global.phase_jump_time + black_pls.fall_time(
        rydberg_global
    )
    assert raw_interval >= rydberg_global.min_duration
    clock = rydberg_global.clock_period
    interval = raw_interval + (
        clock - raw_interval % clock if raw_interval % clock else 0
    )
    assert seq._schedule["ch0"][-1].ti - tf_ == interval
```
Feels a bit like overkill to me but I'm okay with it.

Reply:
This third option tests `Sequence` against `Pulse`, so I think that's equivalent.
I can guess this is hard to test cleanly anyway, because the calculations are complex. The best would probably to keep the first or the second option. And then maybe, add tests inside "test_pulse.py" to actually test `fall_time`, but with simple hard-coded values, instead of calculations (usually, it's better to avoid logics and calculations as much as possible in UTs because they can be as buggy as the tested code).

Reply:
Ok, I added a simple test for `fall_time` as suggested